src/core/include/openvino/core/type/element_type.hpp

```diff
@@ -37,33 +37,34 @@ namespace element {
 /// \brief Enum to define possible element types
 /// \ingroup ov_element_cpp_api
 enum class Type_t {
-    undefined,  //!< Undefined element type
-    dynamic,    //!< Dynamic element type
-    boolean,    //!< boolean element type
-    bf16,       //!< bf16 element type
-    f16,        //!< f16 element type
-    f32,        //!< f32 element type
-    f64,        //!< f64 element type
-    i4,         //!< i4 element type
-    i8,         //!< i8 element type
-    i16,        //!< i16 element type
-    i32,        //!< i32 element type
-    i64,        //!< i64 element type
-    u1,         //!< binary element type
-    u2,         //!< u2 element type
-    u3,         //!< u3 element type
-    u4,         //!< u4 element type
-    u6,         //!< u6 element type
-    u8,         //!< u8 element type
-    u16,        //!< u16 element type
-    u32,        //!< u32 element type
-    u64,        //!< u64 element type
-    nf4,        //!< nf4 element type
-    f8e4m3,     //!< f8e4m3 element type
-    f8e5m2,     //!< f8e5m2 element type
-    string,     //!< string element type
-    f4e2m1,     //!< f4e2m1 element type
-    f8e8m0,     //!< f8e8m0 element type
+    undefined OPENVINO_ENUM_DEPRECATED(
+        "This type is deprecated and will be removed in 2026.0. Use dynamic instead."),  //!< Undefined element type
```

Comment:
should we assign the same values for undefined and dynamic? I.e. undefined = dynamic

Reply:
If same value will assigned then same type info will be used for these types which is different e.g. names.
I can try it but not sure about e.g. the serialization